### PR TITLE
Create class ACEMDTransformation

### PR DESCRIPTION
**Pull request by weifanfrank**
Create class ACEMDTransformation

### CODE REVIEW

File: src/potdata/transformations.py [hunk 1]
```diff
@@ -335,6 +336,61 @@ def run_md(
 
         return structures
 
+class ACEMDTransformation(BaseMDTransformation):
+    """Molecular dynamics transformation using m3gnet and ACE.
```
Comment (mjwen) [inline src/potdata/transformations.py:340]: This is no m3gnet used in this class.

File: src/potdata/transformations.py [hunk 2]
```diff
@@ -21,13 +21,14 @@
     import m3gnet
 except ImportError:
     m3gnet = None
-
+import pyace
```
Comment (mjwen) [inline src/potdata/transformations.py:24]: How about this?

```python
try:
    import pyace
except ImportError:
    pyace = None

File: src/potdata/transformations.py [hunk 3]
```diff
@@ -335,6 +336,61 @@ def run_md(
...
+    ) -> list[Structure]:
+        from ase.io import Trajectory as Trajectory
+        from ase.md.nvtberendsen import NVTBerendsen
+        from ase import units
+        from pymatgen.io.ase import AseAtomsAdaptor
+        from pyace import PyACECalculator
+
+        atoms = AseAtomsAdaptor.get_atoms(structure)
+        # Initialize ACE calculator
+        calc = PyACECalculator("potential.yaml")
```
Comment (mjwen) [inline src/potdata/transformations.py:373]: Never hardcode the filename here. It should be provided from the input argument. 

change line 357 to `potential_filename: Path | str = 'potential.yaml',`
and update this line to `calc = PyACECalculator(potential_filename)`

This way, it is easy to use a potential file in a different directory by providing it to the `potential_filename`.

File: src/potdata/transformations.py [hunk 4]
```diff
@@ -335,6 +336,61 @@ def run_md(
...
+        calc.set_active_set("potential.asi")
```
Comment (mjwen) [inline src/potdata/transformations.py:374]: Similarly for this, no hard code. 
Add a new argument `potential_asi_filename` and use it.

**Review by mjwen — CHANGES_REQUESTED** [review]

File: src/potdata/transformations.py (hunk 4 above)
Comment (weifanfrank) [reply src/potdata/transformations.py]: Done, thanks

File: src/potdata/transformations.py (hunk 3 above)
Comment (weifanfrank) [reply src/potdata/transformations.py]: Done, thanks

File: src/potdata/transformations.py (hunk 2 above)
Comment (weifanfrank) [reply src/potdata/transformations.py]: Done, thanks

File: src/potdata/transformations.py (hunk 1 above)
Comment (weifanfrank) [reply src/potdata/transformations.py]: Done, thanks

File: src/potdata/transformations.py [hunk 5]
```diff
@@ -354,7 +357,8 @@ class ACEMDTransformation(BaseMDTransformation):
     def run_md(
         self,
         structure: Structure,
-        #potential: [Union[Potential, str]],
+        potential_filename: Path | str = 'potential.yaml'
```
Comment (mjwen) [inline src/potdata/transformations.py:360]: missing comma `,` at the end of line. Same for the next line.

**Review by mjwen — CHANGES_REQUESTED** [review]

**Pull request merged**